apps/src/p5lab/redux/animationList.js

```diff
@@ -465,24 +485,22 @@ export function appendCustomFrames(props) {
  * Add a library animation to the project (at the end of the list, unless a spritelab project).
  * @param {!SerializedAnimation} props
  */
-export function addLibraryAnimation(props, skipBackground) {
+export function addLibraryAnimation(props, isSpriteLab) {
   return (dispatch, getState) => {
     const key = createUuid();
     if (getState().pageConstants && getState().pageConstants.isBlockly) {
       dispatch(addAnimationAction(key, props, 0));
     } else {
       dispatch(addAnimationAction(key, props));
     }
-    // if skipBackground, this means we don't want the selected animation to be a background
-    if (!skipBackground || !props.categories.includes('backgrounds')) {
-      dispatch(
-        loadAnimationFromSource(key, () => {
-          dispatch(selectAnimation(key));
-        })
-      );
-    } else {
-      dispatch(loadAnimationFromSource(key, () => {}));
-    }
+    const isSpriteLabBackground =
+      props.categories?.[0] === 'backgrounds' && isSpriteLab;
```

Comment:
nit: can you add a comment explaining that if we are in background mode the categories list will be a single item list containing the category `backgrounds`?

Reply:
The code here controls adding a new animation to the sidebar on the left. This code is used to determine if now we should be selecting a animation or a background based on the category information for the animation that was chosen from the library.

Reply:
👍 I wasn't super clear--I was thinking of a comment explaining why `props.categories?.[0] === 'backgrounds'` will always tell you if it is a background or not. The `[0]` feels a bit magical

Reply:
I updated this to be a an `.includes` to make it a little safer and less magical. In practice, the backgrounds category is mutually exclusive from all other categories, but that's definitely a bit of a fragile system.